msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,348 @@
+## 4.0 App package and store listing
 
-### &#9989; Response times must be reasonable
+App packages must be correctly formatted and include all required information and components.
 
-* **Tabs**. If a response to an action takes more than three seconds, you must provide a loading message or warning.
-* **Bots**. A response to a user command must occur within two seconds. If longer processing is required, your app must display a typing indicator.
-* **Compose extensions**. A response to a user command must occur within five seconds.
+### 4.1 App manifest
+
+* The app manifest must conform to the [latest manifest schema](~/resources/schema/manifest-schema.md).
+* If your app includes a bot or messaging extension, the manifest must be consistent with Bot Framework metadata, including bot name, logo, privacy policy link, and terms of service link.
+* If your app uses Azure Active Directory (Azure AD) for authentication, include the Azure AD ID in the manifest. For more information, see [manifest schema](~/resources/schema/manifest-schema.md#webapplicationinfo).
+
+### 4.2 App icons
+
+* Your app package must include two PNG versions of your app icon: a color icon and an outline icon.
+* Correctly sized and formatted icons must be specified in your app package. The icons must match what's submitted with the store listing metadata. For more information, see the Teams app [icon guidelines](~/concepts/build-and-test/apps-package.md#app-icons).
+
+### 4.3 App description
+
+The app descriptions in your manifest and store must be the same and follow these guidelines:
+
+* Use "Microsoft Teams" on first reference. Subsequent references can be shortened to ”Teams”.
+* Do not abbreviate "Microsoft" (such as "MS" or "MSFT").
+* Avoid unnecessary capitalization and language errors.
+* Follow the [short description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#short-description).
+* Follow the [long description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#long-description).
+* Clearly describe any limitations, conditions, or exceptions to the functionality, features, and deliverables described in the listing and related materials before the customer acquires your offer. The capabilities you declare must relate to the core functions and description of your offer.
+
+### 4.4 Privacy policy
+
+The policy can be specific to your Teams app or an overall policy for all of your services.
+
+* If you use a generic privacy policy template, you must reference "services", "applications", and "platforms" to include your Teams app and your website or service.
+* Must include how you handle user data storage, retention, and deletion. You also must describe the security controls you use for data protection.
+* Must include your contact information.
+* Should not contain broken, beta, or staging URLs.
+
```

Comment:
@heath-hamilton, Please add - 
•	Must not contain any links to AppSource within the privacy policy.